sled-agent/src/rack_setup/plan/service.rs

```diff
@@ -295,13 +295,18 @@ impl Plan {
 
         // Provision internal DNS zones, striping across Sleds.
         let reserved_rack_subnet = ReservedRackSubnet::new(config.az_subnet());
-        let dns_subnets = reserved_rack_subnet.get_dns_subnets();
+        assert!(
+            DNS_REDUNDANCY <= MAX_DNS_REDUNDANCY,
```

Comment:
Tiny nit - since these are both constants, we could use [static_assertions::const_assert](https://docs.rs/static_assertions/latest/static_assertions/macro.const_assert.html) to check this at compile time instead of runtime.